src/fontra/client/web-components/plugin-manager.js

```diff
@@ -0,0 +1,167 @@
+    gap: 0.2rem;
+  }
+
+  .no-plugins {
+    padding: 0.4rem 0.2rem;
+  }
+  `;
+  constructor() {
+    super();
+    this.contentElement = this.shadowRoot.appendChild(html.div());
+    this.contentElement.classList.add("plugin-manager");
+    const observable = new ObservableController({
+      plugins: [],
+    });
+    observable.synchronizeWithLocalStorage("fontra.plugins");
+    this.observable = observable;
+    this.pluginList = new UIList();
+    this.pluginList.setItems(observable.model.plugins);
+    this.pluginList.columnDescriptions = [
+      {
+        key: "address",
+        title: "Github repository",
+      },
+    ];
+    this.render();
+    this.renderPlugins();
+    this.pluginList.addEventListener("listSelectionChanged", async (event) => {
+      this.deleteButton.style.display =
+        this.pluginList.selectedItemIndex === undefined ? "none" : "unset";
+    });
+  }
+
+  async promptAddPlugin(text = "") {
+    const newPluginPrompt = await dialogSetup(
+      "Provide a github handle/repository",
+      "",
+      [
+        { title: "Cancel", resultValue: "no", isCancelButton: true },
+        { title: "Create", resultValue: "ok", isDefaultButton: true },
+      ]
+    );
+    let address = text;
+    newPluginPrompt.setContent(
+      html.input({
+        autofocus: true,
+        value: text,
+        oninput: (event) => {
+          address = event.target.value;
+        },
+      })
+    );
+    const pluginPromptResult = await newPluginPrompt.run();
+    if (pluginPromptResult === "ok") {
+      const newPlugin = { address };
+      const [ok, errorMessage] = await this.validatePlugin(
+        parsePluginBasePath(address)
+      );
+      if (ok) {
+        this.observable.setItem("plugins", [
+          ...this.observable.model.plugins,
+          newPlugin,
+        ]);
+        this.renderPlugins();
+      } else {
+        await dialog("Error", errorMessage, [{ title: "Ok", isDefaultButton: true }]);
+        return this.promptAddPlugin(address);
+      }
+    }
+  }
+
+  async validatePlugin(pluginPath) {
+    if (
+      this.observable.model.plugins.some(
+        ({ address }) => parsePluginBasePath(address) === pluginPath
+      )
+    ) {
+      return [false, "Plugin exists."];
+    }
+    let response;
+    try {
+      response = await fetch(`${pluginPath}/plugin.json`);
+    } catch (e) {
+      return [false, "An error occured when fetching the plugin."];
+    }
+    if (response.status === 404) {
+      return [false, "Plugin not found."];
+    }
+    return [true];
+  }
+
+  renderPlugins() {
+    const plugins = this.observable.model.plugins;
+    this.pluginList.setItems(plugins);
+    if (plugins.length === 0) {
+      this.pluginsEmptyState.style.display = "unset";
+    } else {
+      this.pluginsEmptyState.style.display = "none";
+    }
+  }
+
+  render() {
+    const fragment = document.createDocumentFragment();
+    fragment.appendChild(
+      html.div({ class: "buttons" }, [
```

Comment:
Please use the `add-remove-buttons` web component for this, and place it underneath the list. This is what Fontra does elsewhere (sources list, axes list).